frontend/src/app/api/characters.ts

```diff
@@ -0,0 +1,85 @@
+import { useAccount } from "@gear-js/react-hooks";
+import { graphql } from "gql/gql";
+import {
+  getGraphQLPrimaryKey,
+  useGraphQL,
+} from "app/providers/ReactQuery/useGraphQL";
+import { queryClient } from "app/providers/ReactQuery";
+
+const allCharactersQueryDocument = graphql(/* GraphQL */ `
+  query AllCharacters {
+    characters {
+      id
+      level
+      name
+      owner
+      experience
+      attributes
+    }
+  }
+`);
+
+export const useAllCharacters = () => useGraphQL(allCharactersQueryDocument);
```

Comment:
is it used somewhere?

Reply:
It wasn't. But i think we should use it instead of reading gear state directly (we used to read it because indexer wasn't catching all the messages before)